packages/cli/src/metadataGeneration/transformer/transformer.ts

```diff
@@ -0,0 +1,50 @@
+import * as ts from 'typescript';
+
+import { TypeResolver } from '../typeResolver';
+
+/**
+ * Transformer responsible to transforming native ts node into tsoa type.
+ */
+export abstract class Transformer {
```

Comment:
What do you think about the Transformers registering themselves via a shouldHandle(node) function?

Reply:
Can you give me more details about it? I have no clue about shouldHandle function. Oops.